src/sinks/util/uri.rs

```diff
@@ -1,18 +1,26 @@
-use http::Uri;
+use crate::http::Auth;
+use http::uri::{Authority, Uri};
+use percent_encoding::percent_decode_str;
 use serde::{
     de::{Error, Visitor},
     Deserialize, Deserializer, Serialize, Serializer,
 };
 use std::fmt;
+use std::str::FromStr;
 
 /// A wrapper for `http::Uri` that implements the serde traits.
+/// Authorization credentials, if exist, will be removed from the URI and stored in `auth`.
+/// For example: "http://user:password@example.com".
 #[derive(Default, Debug, Clone)]
-pub struct UriSerde(Uri);
+pub struct UriSerde {
+    pub uri: Uri,
+    #[must_use]
```

Comment:
This is nifty. I hadn't seen that macro before.

Reply:
Thanks for catching! I tried this to see if it would catch when we forgot to call `merge_auth_config`. I've removed it.